bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/discovery/LuxtronikHeatpumpDiscovery.java

```diff
@@ -0,0 +1,231 @@
+        } catch (UnknownHostException e) {
+            logger.debug("Could not build net ip address.", e);
+        }
+        octet = 0;
+    }
+
+    private synchronized String getNextIPAddress(CidrAddress adr) {
+        octet++;
+        octet &= ~ipMask;
+        byte[] octets = adr.getAddress().getAddress();
+        octets[2] += (octet >> 8);
+        octets[3] += octet;
+        String address = "";
+        try {
+            InetAddress iAdr = null;
+            iAdr = InetAddress.getByAddress(octets);
+            address = iAdr.getHostAddress();
+        } catch (UnknownHostException e) {
+            logger.debug("Could not find next ip address.", e);
+        }
+        return address;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Starts the scan. This discovery will:
+     * <ul>
+     * <li>Request this hosts first IPV4 address.</li>
+     * <li>Send a socket request on port 8888 / 8889 to all IPs on the subnet.</li>
+     * <li>The response is then investigated to see if is an answer from a heat pump</li>
+     * </ul>
+     * The process will continue until all addresses are checked, timeout or {@link #stopScan()} is called.
+     */
+    @Override
+    protected void startScan() {
+        if (executorService != null) {
+            stopScan();
+        }
+
+        CidrAddress localAdr = getLocalIP4Address();
+        if (localAdr == null) {
+            stopScan();
+            return;
+        }
+        setupBaseIp(localAdr);
+        CidrAddress baseAdr = baseIp;
+        scanning = true;
+        ExecutorService localExecutorService = Executors.newFixedThreadPool(SCAN_THREADS);
+        executorService = localExecutorService;
+        for (int i = 0; i < addressCount; i++) {
+
+            localExecutorService.execute(() -> {
+                if (scanning && baseAdr != null) {
+                    String ipAdd = getNextIPAddress(baseAdr);
+
+                    boolean detection = (discoverFromIp(ipAdd, 8889) || discoverFromIp(ipAdd, 8888));
+                }
+            });
+        }
+    }
+
+    private boolean discoverFromIp(String ipAdd, int port) {
+        HeatpumpConnector connection = new HeatpumpConnector(ipAdd, port);
+
+        try {
+            connection.read();
+            Integer[] heatpumpValues = connection.getValues();
+            Map<String, Object> properties = ChannelUpdaterJob.getProperties(heatpumpValues);
+            properties.put("port", port);
+
+            String type = properties.get("heatpumpType").toString();
+            ThingTypeUID typeId = LuxtronikHeatpumpBindingConstants.THING_TYPE_HEATPUMP;
+            ThingUID uid = new ThingUID(typeId, type);
+
+            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(type)
+                    .build();
+            thingDiscovered(result);
+
+            return true;
+        } catch (IOException e) {
+            // Ignore, this is the expected behavior.
```

Comment:
A trace log entry could be written.

Reply:
I don't think that would help much. At this point an exception is thrown if no heatpump could be found for the given IP / port. As this is executed for each IP that is discovered in the network it might spam the log with useless entries.

Reply:
That is where a trace log is used for. But if you think it will not help anyone ever, leave it as it is.